src/nvim/os/env.c

```diff
@@ -410,14 +410,14 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
+  char *p;
+  char *pend;
   int vimruntime;
 
 
-  p = (char_u *)os_getenv((char *)name);
+  p = (char *)os_getenv(name);
```

Comment:
> the cast to (char_u) that was there *seemed to me to do the same thing, so it appeared to me like my changes were a lateral move, rather than a step back.

Mmmm. You're right. I hadn't realized incorrect cast to `char_u *` was already there before.

That's something that should be fixed sooner or later, though, so, if not too complicated, advice to fix it remains. Apart  from the "no step back" rule, we tend to apply the "step forward if possible". ;-)

But if you feel more comfortable leaving that for next PR, as you suggested, I'm ok with it, as as you said, it doesn't break anything that wasn't broken before.